app/controllers/application_controller.rb

```diff
@@ -75,6 +75,24 @@ def find_platform(param = :id)
   def find_project
     @project = Project.find_with_includes!(params[:platform], params[:name], [:repository, :versions])
     @color = @project.color
+  rescue ActiveRecord::RecordNotFound
+    raise if params[:name].blank? || params[:platform].downcase != "go"
```

Comment:
There's also `app/controllers/api/application_controller.rb:find_proect`. I wonder if there's a way we could have this handled in `Project.find_with_includes`? It looks like that already does some normalization. And we could then do in a way to allow resolving names for other package managers in the future

Reply:
Could definitely move it there

Reply:
I think this is the only `find_project` - `Api::ApplicationController` doesn't seem to have one.